macosx/QuickLookExtension/PreviewProvider.mm

```diff
@@ -0,0 +1,290 @@
+
+    NSUInteger const width = 32;
+    [htmlString appendFormat:@"<h2><img class=\"icon\" src=\"%@\" width=\"%ld\" height=\"%ld\" />%@</h2>",
+                             generateIconData(fileType, width, attachments),
+                             width,
+                             width,
+                             name];
+
+    NSString* fileSizeString = [NSString stringForFileSize:metainfo.total_size()];
+    if (is_multifile)
+    {
+        NSString* fileCountString = [NSString
+            localizedStringWithFormat:NSLocalizedStringFromTableInBundle(@"%lu files", nil, bundle, "quicklook file count"), n_files];
+        fileSizeString = [NSString stringWithFormat:@"%@, %@", fileCountString, fileSizeString];
+    }
+    [htmlString appendFormat:@"<p>%@</p>", fileSizeString];
+
+    auto const date_created = metainfo.date_created();
+    NSString* dateCreatedString = date_created > 0 ?
+        [NSDateFormatter localizedStringFromDate:[NSDate dateWithTimeIntervalSince1970:date_created] dateStyle:NSDateFormatterLongStyle
+                                       timeStyle:NSDateFormatterShortStyle] :
+        nil;
+    auto const& creator = metainfo.creator();
+    NSString* creatorString = !std::empty(creator) ? @(creator.c_str()) : nil;
+    if ([creatorString isEqualToString:@""])
+    {
+        creatorString = nil;
+    }
+    NSString* creationString = nil;
+    if (dateCreatedString && creatorString)
+    {
+        creationString = [NSString
+            stringWithFormat:NSLocalizedStringFromTableInBundle(@"Created on %@ with %@", nil, bundle, "quicklook creation info"),
+                             dateCreatedString,
+                             creatorString];
+    }
+    else if (dateCreatedString)
+    {
+        creationString = [NSString
+            stringWithFormat:NSLocalizedStringFromTableInBundle(@"Created on %@", nil, bundle, "quicklook creation info"), dateCreatedString];
+    }
+    else if (creatorString)
+    {
+        creationString = [NSString
+            stringWithFormat:NSLocalizedStringFromTableInBundle(@"Created with %@", nil, bundle, "quicklook creation info"), creatorString];
+    }
+    if (creationString)
+    {
+        [htmlString appendFormat:@"<p>%@</p>", creationString];
+    }
+
+    auto const& commentStr = metainfo.comment();
+    if (!std::empty(commentStr))
+    {
+        NSString* comment = @(commentStr.c_str());
+        if (![comment isEqualToString:@""])
+            [htmlString appendFormat:@"<p>%@</p>", comment];
+    }
+
+    NSMutableArray* lists = [NSMutableArray array];
+
+    auto const n_webseeds = metainfo.webseed_count();
+    if (n_webseeds > 0)
+    {
+        NSMutableString* listSection = [NSMutableString string];
+        [listSection appendString:@"<table>"];
+
+        NSString* headerTitleString = n_webseeds == 1 ?
+            NSLocalizedStringFromTableInBundle(@"1 Web Seed", nil, bundle, "quicklook web seed header") :
```

Comment:
Ditto about the amount pluralisation. There are also more strings like that below

Reply:
https://github.com/transmission/transmission/pull/7213#discussion_r1885765181